app/webpacker/controllers/terms_and_conditions_controller.js

```diff
@@ -11,15 +12,23 @@ export default class extends ApplicationController {
   remove(event) {
     let confirmation = confirm(this.messageValue);
     if (confirmation) {
-      // remove shebang present in the enterprise/edit url
-      // because of the shebang rails cannot parse the url
-      // as it is not valid
       location.hash = "";
-
       this.stimulate(
         "EnterpriseEdit#remove_terms_and_conditions",
         event.target
       );
     }
   }
+
+  stopEvent(event) {
+    event.preventDefault();
+  }
+
+  add() {
+    let input = document.querySelector("#enterprise_terms_and_conditions");
+    input.removeAttribute("data-action");
```

Comment:
I'll be first to admit that I think the code is a bit hacky. I couldn't find another way to add this functionality.

The modal window needs to open after the `browse` button is clicked by the user and only after the `confirm` button in the modal is clicked should the default behaviour (the file selection window open) continue.

In angular this behaviour is achieved through here:
```coffee
"app/assets/javascripts/admin/enterprises/directives/terms_and_conditions_warning.js.coffee"

     # When the user presses continue in the warning modal, we open the browser file input dialog
    scope.continue = ->
      scope.template.dialog('close')
      $rootScope.$evalAsync()
```

And I'm reading the whole thing as `event.preventDefault()` then if the user presses `continue` -> `event.continueDefault()` but there's no such thing in native js, which is why I came to this solution. 

Also I have a half a second's delay before clicking the input because when I didn't have that the file selection window wouldn't show. Must be something in the close-modal code that prevents it.

Reply:
I should re-read the code, and test myself. Will give a try.

Reply:
Sure. Take your time JB.

btw, below where you have commented on the segment of code

```ruby
  add() {
    let input = document.querySelector("#enterprise_terms_and_conditions");
    input.removeAttribute("data-action");
```

This line `input.removeAttribute("data-action");` is just removing the event handlers from the `input`. I am not sure that it 100% removes the event handlers but after the `data-action` attributes are removed the corresponding `stimulus` actions don't run which is why you can click on the input and have the file selection window pop up. In contrast to, having the `terms and conditions warning` modal popup.

Reply:
> I'll be first to admit that I think the code is a bit hacky. I couldn't find another way to add this functionality.

I'm wondering if we can simplify the T&Cs upload model, @openfoodfoundation/train-drivers-product-owners. Can we show that information as normal text within the form so that you can read it any time instead of just popping it up when you try to upload a file?

I'm sure that once you read that warning once, it's annoying to have it getting in the way when you want to upload the next file. It's very unexpected to have a modal pop up like that. A bit of that is good to make sure that people don't overlook it but I think that it's a bit much. I'm hoping that we can improve UX and simplify code at the same time.

Reply:
Conversation moved to Slack [here](https://openfoodnetwork.slack.com/archives/C018S101R4J/p1665983975954109).